app/promoCodeFirstRunStorage.js

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const electron = require('electron')
+const path = require('path')
+const fs = require('fs')
+
+const promoCodeFileRelativeUserDataPath = 'promoCode'
+
+function getPromoCodeFileAbsolutePath () {
+  return path.join(electron.app.getPath('userData'), promoCodeFileRelativeUserDataPath)
+}
+
+module.exports = {
+  writeFirstRunPromoCodeSync: promoCode => {
+    const promoCodeOutputPath = getPromoCodeFileAbsolutePath()
+        // write promo code so state can access it
+    fs.writeFileSync(promoCodeOutputPath, promoCode)
+  },
+  readFirstRunPromoCode: () => new Promise((resolve, reject) => {
```

Comment:
@NejcZdovc this function can be used to read the code so the path is kept in one place. Let me know if it meets your requirements.

Reply:
lgtm but curious why this is written to a separate file instead of as a property in appState

Reply:
oh, i guess this is needed on mac because the equivalent is done in a postinstall script and you probably want windows/mac to work the same way

Reply:
Iw as wondering the same as @diracdeltas but her reasoning makes sense.

Reply:
yeah was interfacing with the spec + code that was already done in #12798 to read from this file - consistency was an ok reason to not suggest we change it